client/my-sites/site-settings/jetpack-module-toggle.jsx

```diff
@@ -38,16 +42,32 @@ class JetpackModuleToggle extends Component {
 		isJetpackSite: PropTypes.bool,
 		activateModule: PropTypes.func,
 		deactivateModule: PropTypes.func,
+		path: PropTypes.string,
+		userID: PropTypes.number,
+		blogID: PropTypes.number,
 	};
 
 	handleChange = () => {
 		if ( ! this.props.checked ) {
+			this._recordTracksEvent( 'calypso_jetpack_module_toggle_activated' );
 			this.props.activateModule( this.props.siteId, this.props.moduleSlug );
 		} else {
+			this._recordTracksEvent( 'calypso_jetpack_module_toggle_deactivated' );
 			this.props.deactivateModule( this.props.siteId, this.props.moduleSlug );
 		}
 	};
 
+	_recordTracksEvent = name => {
+		const tracksProps = {
+			path: this.props.path,
+			module: this.props.moduleSlug,
+			userid: this.props.userID,
+			blogid: this.props.blogID,
```

Comment:
Site and user are not provided by the `analytics` module. I checked this by looking at the props in the debug output when I didn't pass them in manually.

I've called it `blogid` because the FG page on tracks event validation says that events that happen in a site context should have `blogid` to be considered valid.

Reply:
> Site and user are not provided by the analytics module.

I think it does provide the [user id](https://github.com/Automattic/wp-calypso/blob/a5c00596e329b79f185803328c3af8caade233bb/client/lib/analytics/index.js#L115). Apparently it may also provide the [blog id](https://github.com/Automattic/wp-calypso/blob/116678c2b3f7c4b0844fd181da8c96d28f3f6417/client/lib/analytics/super-props.js#L25) but it's not clear when exactly (at least to me).

> I've called it blogid because the FG page on tracks event validation says that events that happen in a site context should have blogid to be considered valid.

Good to know, I wasn't aware of that.

Reply:
It does provide the user ID via `_ui`, and the blog ID is set if a `path` prop is passed with something like `:site` in it, I think to indicate that it is in a site context.